packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -0,0 +1,1446 @@
+    _valueIndicatorAnimation.addListener(markNeedsPaint);
+    _enableAnimation.addListener(markNeedsPaint);
+    _state.startPositionController.addListener(markNeedsPaint);
+    _state.endPositionController.addListener(markNeedsPaint);
+  }
+
+  @override
+  void detach() {
+    _overlayAnimation.removeListener(markNeedsPaint);
+    _valueIndicatorAnimation.removeListener(markNeedsPaint);
+    _enableAnimation.removeListener(markNeedsPaint);
+    _state.startPositionController.removeListener(markNeedsPaint);
+    _state.endPositionController.removeListener(markNeedsPaint);
+    super.detach();
+  }
+
+  double _getValueFromVisualPosition(double visualPosition) {
+    switch (textDirection) {
+      case TextDirection.rtl:
+        return 1.0 - visualPosition;
+      case TextDirection.ltr:
+        return visualPosition;
+    }
+    return null;
+  }
+
+  double _getValueFromGlobalPosition(Offset globalPosition) {
+    final double visualPosition = (globalToLocal(globalPosition).dx - _trackRect.left) / _trackRect.width;
+    return _getValueFromVisualPosition(visualPosition);
+  }
+
+  double _discretize(double value) {
+    double result = value.clamp(0.0, 1.0);
+    if (isDiscrete) {
+      result = (result * divisions).round() / divisions;
+    }
+    return result;
+  }
+
+  RangeValues _discretizeRangeValues(RangeValues values) {
+    return RangeValues(_discretize(values.start), _discretize(values.end));
+  }
+
+  void _startInteraction(Offset globalPosition) {
+    final double tapValue = _getValueFromGlobalPosition(globalPosition).clamp(0.0, 1.0);
+    _lastThumbSelection = thumbSelector(textDirection, values, tapValue, _thumbSize, size, 0);
+
+    if (_lastThumbSelection != null) {
+      _active = true;
+      // We supply the *current* values as the start locations, so that if we have
+      // a tap, it consists of a call to onChangeStart with the previous value and
+      // a call to onChangeEnd with the new value.
+      final RangeValues currentValues = _discretizeRangeValues(values);
+      RangeValues newValues;
+      if (_lastThumbSelection == Thumb.start) {
+        newValues = RangeValues(tapValue, currentValues.end);
+      } else if (_lastThumbSelection == Thumb.end) {
+        newValues = RangeValues(currentValues.start, tapValue);
+      }
+      _updateLabelPainter(_lastThumbSelection);
+
+      if (onChangeStart != null) {
+        onChangeStart(currentValues);
+      }
+
+      onChanged(_discretizeRangeValues(newValues));
+
+      _state.overlayController.forward();
+      if (showValueIndicator) {
+        _state.valueIndicatorController.forward();
+        _state.interactionTimer?.cancel();
+        _state.interactionTimer =
+            Timer(_minimumInteractionTime * timeDilation, () {
+              _state.interactionTimer = null;
+              if (!_active &&
+                  _state.valueIndicatorController.status ==
+                      AnimationStatus.completed) {
```

Comment:
I would say this is more readable as:

```
if (!_active &&
    _state.valueIndicatorController.status == AnimationStatus.completed) {
```

More of a nit though.

Reply:
Put it on one line